usr/lib/common/asn1.c

```diff
@@ -2642,8 +2642,15 @@ CK_RV der_encode_ECPrivateKey(CK_BBOOL length_only,
             goto error;
        }
 
-        ber_encode_CHOICE(FALSE, 1, &buf2, &len, (CK_BYTE *)val->bv_val,
-                          val->bv_len);
+        rc = ber_encode_CHOICE(FALSE, 1, &buf2, &len, (CK_BYTE *)val->bv_val,
```

Comment:
Do we need to do the same above in line 2567 ?

Reply:
Good catch! 

I guess Coverty did not flag that one because it's a `length_only = TRUE` operation, so it won't set `buf2` anyway, but just update `len`. Also `buf2` is not touched/used afterwards. 
Nevertheless, we should also check the return code here, otherwise we might continue with a wrong length if that `ber_encode_CHOICE` did fail.

Will fix.